src/crimson/os/seastore/seastore.cc

```diff
@@ -813,31 +838,50 @@ SeaStore::tm_ret SeaStore::_setattrs(
   logger().debug("{} onode={}",
                 __func__, *onode);
   auto& layout = onode->get_mutable_layout(*ctx.transaction);
-  for (auto& [key, val] : aset) {
+  for (auto it = aset.begin(); it != aset.end();) {
+    auto& key = it->first;
+    auto& val = it->second;
+    logger().debug("{} onode={}, key={}",
+                  __func__, *onode, key);
     if (key == OI_ATTR) {
       if (__builtin_expect(
-          val.length() > onode_layout_t::MAX_OI_LENGTH,
-          false)) {
-        logger().error("{} onode={} oi attr too long!");
-        return crimson::ct_error::input_output_error::make();
+          val.length() <= onode_layout_t::MAX_OI_LENGTH,
+          true)) {
+        layout.oi_size = val.length();
+        __builtin_memcpy(&layout.oi[0], val.c_str(), val.length());
+        it = aset.erase(it);
+        continue;
+      } else {
+        layout.oi_size = 0;
```

Comment:
Why change to silent the error when the length is too long?

Reply:
oi and ss attrs that too long to inline with onode will be stored in the omap tree whose root is xattr_root

Reply:
OK.